### PR TITLE
JBPM-5922: Cannot save a diagram until it's valid.

### PR DESCRIPTION
**Pull request by romartin**
Hey @manstis 

That's a small fix for validations. See [the JIRA](https://issues.jboss.org/browse/JBPM-5922) ticket.

Thanks!

### CODE REVIEW

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/notification/NotificationMessageUtils.java [hunk 1]
```diff
@@ -58,15 +59,21 @@ public static String getCanvasValidationsErrorMessage(final ClientTranslationSer
...
 
     public static String getBeanValidationMessage(final ClientTranslationService translationService,
                                                   final ModelBeanViolation violation) {
-        return OPEN_COMMENT + violation.getPropertyPath().toString() + CLOSE_COMMENT
+        return getViolationTypeMessage(violation) +
+                OPEN_COMMENT + violation.getPropertyPath() + CLOSE_COMMENT
                 + violation.getMessage();
     }
 
+    private static String getViolationTypeMessage(final
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/notification/NotificationMessageUtils.java]: Formatting... but let's not worry for the impending 7.0.0.Final....

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/graph/processing/traverse/tree/TreeWalkTraverseProcessorImplTest.java [hunk 2]
```diff
@@ -168,4 +168,50 @@ public void testTraverseGraph2UsingRootNode() {
         verify(callback,
                times(1)).endGraphTraversal();
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTraverseGraph1AndPendingEdges() {
+        final TestingGraphInstanceBuilder.TestGraph1 result =
+                TestingGraphInstanceBuilder.newGraph1(graphTestHandler);
+        // Create and add a new incomming view edge into the intermediate node,
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/graph/processing/traverse/tree/TreeWalkTraverseProcessorImplTest.java]: Typo "inco**m**ing..." but let's not worry about for 7.0.x

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/rule/violations/DefaultRuleViolationsTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    @Before
+    public void setup() throws Exception {
+        when(violationErr.getViolationType()).thenReturn(Violation.Type.ERROR);
+        when(violationWarn.getViolationType()).thenReturn(Violation.Type.WARNING);
+        when(violationInfo.getViolationType()).thenReturn(Violation.Type.INFO);
+    }
+
+    @Test
+    public void test() {
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/rule/violations/DefaultRuleViolationsTest.java]: A nice descriptive test method name ;-)

Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/rule/violations/DefaultRuleViolationsTest.java]: You could perhaps split this into three different tests for the different types.

Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/test/java/org/kie/workbench/common/stunner/core/rule/violations/DefaultRuleViolationsTest.java]: finally only changed the name - it's a test that returns the violations for a concrete type, so I think that's fine a single test case that add violations for the 3 types and just checks it returns the right one for each predicate....

**Comment by romartin** [comment]
@manstis  new commit with the updates added. thx!!

**Pull request merged**